Build failuar on CRAN Windows builder

I sent 6be3f1fe2300aa24ac1b61e8569b0f1c45e61fad to CRAN but it does not pass the incoming checks automatically.

```log
* installing *source* package 'prqlr' ...
** using staged installation

--------------------------- [RUST FOUND] ---------------------------
cargo 1.76.0 (c84b36747 2024-01-18)

rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: x86_64-pc-windows-gnu
release: 1.76.0
LLVM version: 17.0.6
--------------------------------------------------------------------

** libs
using C compiler: 'gcc.exe (GCC) 12.3.0'
rm -Rf "prqlr.dll" "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target/x86_64-pc-windows-gnu/release/libprqlr.a" "init.o"
gcc  -I"D:/RCompile/recent/R/include" -DNDEBUG     -I"d:/rtools43/x86_64-w64-mingw32.static.posix/include"     -pedantic -Wstrict-prototypes -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c init.c -o init.o
if [ -f "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/../tools/libprqlr.a" ]; then \
	mkdir -p "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target/x86_64-pc-windows-gnu/release" ; \
	mv "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/../tools/libprqlr.a" "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target/x86_64-pc-windows-gnu/release/libprqlr.a" ; \
	exit 0; \
fi && \
if [ -f "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/vendor.tar.xz" ]; then \
	mkdir -p "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/vendor" && \
	d:/rtools43/usr/bin/tar --extract --xz --file "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/vendor.tar.xz" -C "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/vendor" && \
	mkdir -p "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/.cargo" && \
	cp "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/vendor-config.toml" "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/.cargo/config.toml"; \
fi && \
if [ "" != "true" ]; then \
	export CARGO_HOME="/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/.cargo"; \
	export CARGO_BUILD_JOBS=2; \
fi && \
	export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="x86_64-w64-mingw32.static.posix-gcc.exe" && \
	cargo build --lib --manifest-path="/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/Cargo.toml" --target-dir "/d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target" --target="x86_64-pc-windows-gnu" \
		--profile="release" --features=""
   Compiling proc-macro2 v1.0.71
   Compiling unicode-ident v1.0.12
   Compiling cc v1.0.83
error: linking with `x86_64-w64-mingw32.static.posix-gcc.exe` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32.static.posix-gcc.exe" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "D:\\temp\\rustcS2JgtJ\\symbols.o" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\build\\proc-macro2-999638963ec1b075\\build_script_build-999638963ec1b075.build_script_build.822b784807005e7d-cgu.0.rcgu.o" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\build\\proc-macro2-999638963ec1b075\\build_script_build-999638963ec1b075.build_script_build.822b784807005e7d-cgu.1.rcgu.o" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\build\\proc-macro2-999638963ec1b075\\build_script_build-999638963ec1b075.build_script_build.822b784807005e7d-cgu.2.rcgu.o" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\build\\proc-macro2-999638963ec1b075\\build_script_build-999638963ec1b075.24epqwbzz8zcwbci.rcgu.o" "-L" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\deps" "-L" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd-efaf1f98469bce6b.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libpanic_unwind-d359dc6e62e55626.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libobject-3c268a13c68ea8e2.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libmemchr-9f58f527e2b91dae.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libaddr2line-c77bb5c52ef40ec2.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libgimli-351e9eebebf3246b.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_demangle-98eb1ba61e5b7801.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd_detect-7e82d8ed3fcb6f19.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libhashbrown-b42813017cfa0a7e.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_alloc-ebd56e19753fc0ec.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libminiz_oxide-633cedffad10403c.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libadler-65d6646cd32deaaa.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libunwind-24b4e764e2c1853b.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcfg_if-9fcf4cdda0a238d5.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-56337887c3a2ba98.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc-257f356590128be9.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_core-4a3f13ff983c2d26.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcore-76b1656df760b6c0.rlib" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-6c9e2f948258138b.rlib" "-Wl,-Bdynamic" "-lkernel32" "-ladvapi32" "-lbcrypt" "-lkernel32" "-lntdll" "-luserenv" "-lws2_32" "-lkernel32" "-lws2_32" "-lkernel32" "-lntdll" "-lkernel32" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-o" "D:/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target\\release\\build\\proc-macro2-999638963ec1b075\\build_script_build-999638963ec1b075.exe" "-Wl,--gc-sections" "-no-pie" "-nodefaultlibs" "C:\\Users\\CRAN\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: D:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: cannot find -lgcc_eh: No such file or directory
          collect2.exe: error: ld returned 1 exit status
          

error: could not compile `proc-macro2` (build script) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
make: *** [Makevars.win:23: /d/temp/RtmpEDufiR/R.INSTALLa42c41d937a6/prqlr/src/rust/target/x86_64-pc-windows-gnu/release/libprqlr.a] Error 101
ERROR: compilation failed for package 'prqlr'
* removing 'd:/RCompile/CRANincoming/R-devel/lib/prqlr'
```

It seems that the same error occurred on #184.

@yutannihilation Sorry for bothering you, but is it possible that the lines below that you removed on #252 are still needed?

https://github.com/PRQL/prqlc-r/blob/97fa7ce6cece4c36634f0f9cc8b9d117ca704efb/src/Makevars.win#L23-L30

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -45,6 +45,7 @@ $(STATLIB):
 		export CARGO_BUILD_JOBS=2; \
 	fi && \
 		export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="$(CARGO_LINKER)" && \
+		export LIBRARY_PATH="$(LIBRARY_PATH);$(LIBGCC_MOCK_DIR)" && \
 		cargo build --lib --manifest-path="$(CURDIR)/rust/Cargo.toml" --target-dir "$(TARGET_DIR)" --target="$(TARGET)" \
 			--profile="$(PRQLR_PROFILE)" --features="$(PRQLR_FEATURES)"
 

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -18,6 +18,7 @@ $(SHLIB): $(STATLIB)
 
 CARGOTMP = $(CURDIR)/.cargo
 VENDOR_DIR = $(CURDIR)/rust/vendor
+LIBGCC_MOCK_DIR = $(CURDIR)/libgcc_mock
 
 $(STATLIB):
 	# `rustc` adds `-lgcc_eh` flags to the compiler, but Rtools' GCC doesn't have
@@ -26,7 +27,7 @@ $(STATLIB):
 	#
 	# For more details, please refer to
 	# https://github.com/r-windows/rtools-packages/blob/2407b23f1e0925bbb20a4162c963600105236318/mingw-w64-gcc/PKGBUILD#L313-L316
-	mkdir -p "$(LIBDIR)/libgcc_mock" && touch "$(LIBDIR)/libgcc_mock/libgcc_eh.a"
+	mkdir -p "$(LIBGCC_MOCK_DIR)" && touch "$(LIBGCC_MOCK_DIR)/libgcc_eh.a"
 
 	if [ -f "$(CURDIR)/../tools/$(LIBNAME)" ]; then \
 		mkdir -p "$(LIBDIR)" ; \
@@ -52,7 +53,7 @@ $(STATLIB):
 	fi
 
 C_clean:
-	rm -Rf "$(SHLIB)" "$(STATLIB)" "$(OBJECTS)"
+	rm -Rf "$(SHLIB)" "$(STATLIB)" "$(OBJECTS)" "$(LIBGCC_MOCK_DIR)"
 
 clean:
-	rm -Rf "$(SHLIB)" "$(STATLIB)" "$(OBJECTS)" "$(TARGET_DIR)"
+	rm -Rf "$(SHLIB)" "$(STATLIB)" "$(OBJECTS)" "$(TARGET_DIR)" "$(LIBGCC_MOCK_DIR)"

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -20,6 +20,14 @@ CARGOTMP = $(CURDIR)/.cargo
 VENDOR_DIR = $(CURDIR)/rust/vendor
 
 $(STATLIB):
+	# `rustc` adds `-lgcc_eh` flags to the compiler, but Rtools' GCC doesn't have
+	# `libgcc_eh` due to the compilation settings. So, in order to please the
+	# compiler, we need to add empty `libgcc_eh` to the library search paths.
+	#
+	# For more details, please refer to
+	# https://github.com/r-windows/rtools-packages/blob/2407b23f1e0925bbb20a4162c963600105236318/mingw-w64-gcc/PKGBUILD#L313-L316
+	mkdir -p "$(LIBDIR)/libgcc_mock" && touch "$(LIBDIR)/libgcc_mock/libgcc_eh.a"
+
 	if [ -f "$(CURDIR)/../tools/$(LIBNAME)" ]; then \
 		mkdir -p "$(LIBDIR)" ; \
 		mv "$(CURDIR)/../tools/$(LIBNAME)" "$(STATLIB)" ; \

